serde_json_path_core/src/spec/integer.rs

```diff
@@ -22,19 +22,19 @@ const MAX: i64 = 9_007_199_254_740_992 - 1;
 /// The minimum allowed value (-2^53) + 1
 const MIN: i64 = -9_007_199_254_740_992 + 1;
 
+fn check_i64(v: i64) -> bool {
+    (MIN..=MAX).contains(&v)
+}
+
 impl Integer {
     fn try_new(value: i64) -> Result<Self, IntegerError> {
-        if !(MIN..=MAX).contains(&value) {
-            Err(IntegerError::OutOfBounds)
-        } else {
+        if check_i64(value) {
             Ok(Self(value))
+        } else {
+            Err(IntegerError::OutOfBounds)
         }
     }
 
-    fn check(&self) -> bool {
-        (MIN..=MAX).contains(&self.0)
-    }
-
     /// Produce an [`Integer`] with the value 0
     pub fn zero() -> Self {
```

Comment:
Since `Integer` has no `&mut self` methods, would it make sense to define 0 as constant, e.g. `Integer::ZERO`?

But I am here not sure either if there are any conventions.

Reply:
Using a constant for this definitely makes more sense - I will try that.